### PR TITLE
refactor(route): wsj news

### PR DESCRIPTION
**Pull request by NavePnow**
## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

```routes
/wsj/en-us
/wsj/en-us/opinion
/wsj/zh-cn
/wsj/zh-cn/opinion
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [x] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [x] 反爬/频率限制 anti-bot or rate limit?
  - [x] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note

Why do I refactor this route?
1. The official RSS provided by wsj.com has less update frequency than the website. The newest tech news is even in February. Thus, I refactor the route by directly scraping the information from the website instead of the official RSS.
2. The official RSS doesn't cover all contents. For example, there are only six categories provided by the official RSS, but the website itself has up to fourteen categories. Besides, the Chinese sites only have one RSS feed, which is messy.
3. I create a utils.js to decouple the workflow and use tiny-async-pool to get rid of rate limit.

### CODE REVIEW

File: lib/v2/wsj/utils.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
+const asyncPool = require('tiny-async-pool');
+const cheerio = require('cheerio');
+const got = require('@/utils/got');
+const { parseDate } = require('@/utils/parse-date');
+
+const chromeMobileUserAgent = 'Mozilla/5.0 (Linux; Android 7.0; SM-G892A Build/NRD90M; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/67.0.3396.87 Mobile Safari/537.36';
```
Comment (TonyRL) [inline lib/v2/wsj/utils.js:6]: const useragent = require('@/utils/rand-user-agent')({ browser: 'chrome', os: 'android', device: 'mobile' });
Please use an updated randomise UA string

File: lib/v2/wsj/utils.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        // Fetch the AMP version
+        const url = item.link.replace(/(?<=^https:\/\/\w+\.wsj\.com)/, '/amp');
+        const response = await got({
+            url,
+            method: 'get',
+            headers: {
+                'User-Agent': chromeMobileUserAgent,
+            },
+        });
+        const html = response.body;
```
Comment (TonyRL) [inline lib/v2/wsj/utils.js:18]:
```suggestion
        const html = response.data;
```

File: lib/v2/wsj/utils.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            $(cover).remove();
+        }
+
+        // Summary
+        const summary = $('head > meta[name="description"]').attr('content');
+
+        // Metadata (categories & updatedAt)
+        const updatedAt = $('meta[itemprop="dateModified"]').attr('content');
+        const publishedAt = $('meta[itemprop="datePublished"]').attr('content');
+        const author = $('.author > a[rel="author"]').html();
```
Comment (TonyRL) [inline lib/v2/wsj/utils.js:36]: `author` accept a plain string only

File: docs/en/traditional-media.md [hunk 4]
```diff
@@ -678,7 +678,17 @@ Provides all of the articles by the specified New York Times author.
 
 ### News
 
-<RouteEn author="oppilate" example="/wsj/en-us/opinion" path="/wsj/:lang/:category?" :paramsDesc="['Language, `en-us`, `zh-cn`, `zh-tw` are supported', 'Category, only supported in `en-us`. Supports `opinion`, `world_news`, `us_bussiness`, `market_news`, `technology`, `lifestyle`.']">
+<RouteEn author="oppilate NavePnow" example="/wsj/en-us/opinion" path="/wsj/:lang/:category?" :paramsDesc="['Language, `en-us`、`zh-cn`、`zh-tw`', 'Category. See below']">
```
Comment (TonyRL) [inline docs/en/traditional-media.md:681]:
```suggestion
<RouteEn author="oppilate NavePnow" example="/wsj/en-us/opinion" path="/wsj/:lang/:category?" :paramsDesc="['Language, `en-us`, `zh-cn`, `zh-tw`', 'Category. See below']">
```
Comma is used to separate list items, there is no `、` in English
https://www.grammarly.com/blog/comma/

**Review by kmk142789 — APPROVED** [review]

**Pull request merged**